CMake errors when using -DBUILD_SHARED_LIBS=OFF after #1178

Hi,

It seems sth went off after #1178 and now I'm getting these errors in my project:

```CMake
CMake Error: install(EXPORT "ctranslate2Targets" ...) includes target "ctranslate2" which requires target "spdlog_header_only" that is not in any export set.
CMake Error in ext/CTranslate2/CMakeLists.txt:
  export called with target "ctranslate2" which requires target
  "spdlog_header_only" that is not in any export set.


CMake Error in ext/CTranslate2/CMakeLists.txt:
  export called with target "ctranslate2" which requires target
  "cpu_features" that is not in any export set.
```

Building from command line seems OK, so do I have to change anything in my project's CMakeLists.txt?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -558,24 +558,31 @@ write_basic_package_version_file(
   COMPATIBILITY AnyNewerVersion
 )
 
-export(EXPORT ${PROJECT_NAME}Targets
-  FILE "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/${PROJECT_NAME}Targets.cmake"
-  NAMESPACE CTranslate2::
-)
+if(BUILD_SHARED_LIBS)
+  export(EXPORT ${PROJECT_NAME}Targets
+    FILE "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/${PROJECT_NAME}Targets.cmake"
+    NAMESPACE CTranslate2::
+  )
+endif()
+
 configure_file(cmake/${PROJECT_NAME}Config.cmake
   "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/${PROJECT_NAME}Config.cmake"
   COPYONLY
 )
 
 set(ConfigPackageLocation lib/cmake/${PROJECT_NAME})
-install(EXPORT ${PROJECT_NAME}Targets
-  FILE
-    ${PROJECT_NAME}Targets.cmake
-  NAMESPACE
-    CTranslate2::
-  DESTINATION
-    ${ConfigPackageLocation}
-)
+
+if(BUILD_SHARED_LIBS)
+  install(EXPORT ${PROJECT_NAME}Targets
+    FILE
+      ${PROJECT_NAME}Targets.cmake
+    NAMESPACE
+      CTranslate2::
+    DESTINATION
+      ${ConfigPackageLocation}
+  )
+endif()
+
 install(
   FILES
     cmake/${PROJECT_NAME}Config.cmake

